tools/cfn2ts/lib/codegen.ts

```diff
@@ -514,15 +520,26 @@ export default class CodeGenerator {
     this.code.closeBlock();
   }
 
-  private emitProperty(context: genspec.CodeName, propName: string, spec: schema.Property, additionalDocs: string): string {
-    const question = spec.Required ? '' : '?';
-    const javascriptPropertyName = genspec.cloudFormationToScriptName(propName);
+  private emitProperty(props: EmitPropertyProps): string {
```

Comment:
this is the bulk of m code approach question. I have taken the least change path in this design. However, I debated a larger refactor here where we pull out the concept of this property set that is associated to a resource into an object. Then we would call `renderInterfaceProps()` and `renderClassProps()`. This is a much larger refactor because of where these code generation activities are called and connected. Specifically the validators and mappers are coupled to the current design and would also need refactored.

So I'm seeking feedback on if the investment in a better OO model design here is worth the effort?